src/terminator/intrinsic.rs

```diff
@@ -383,8 +383,17 @@ impl<'a, 'tcx> EvalContext<'a, 'tcx> {
             }
 
             "transmute" => {
+                let src_ty = substs.type_at(0);
                 let dest_ty = substs.type_at(1);
-                self.write_value(arg_vals[0], dest, dest_ty)?;
+                let (_, src_align) = self.size_and_align_of_dst(src_ty, arg_vals[0])?;
+                let (size, dest_align) = self.size_and_align_of_dst(dest_ty, arg_vals[0])?;
```

Comment:
Good point. For some reason I was mentally expanding the "dst" part of `size_and_align_of_dst()` into "destination", not "dynamically sized type".